### PR TITLE
Added support for customizing storage class

### LINKED ISSUE #19: Support for Customizing StorageClass
**Is your feature request related to a problem? Please describe.**
Add support for customizing StorageClass for Cassandra PVCs

**Describe the solution you'd like**
Adding Support  StorageClass to support all platforms

**Describe alternatives you've considered**
NA
**Additional context**
NA

### PR DESCRIPTION
**Pull request by RiflerRick**
Fixes #19 

- updated README for custom storage classes

### CODE REVIEW

File: playbook.yaml [hunk 1]
```diff
@@ -26,14 +26,15 @@
   roles:
     - { role: validate-input, tags: ['validate-input'] }
     - { role: validate-input-apigee-control-plane, tags: ['validate-input', 'validate-input-apigee'] }
-    - { role: validate-input-k8s-cluster, tags: ['validate-input', 'validate-input-k8s'] }
+    - { role: validate-input-k8s-cluster, vars: { custom_cassandra_storage_class: '{{ overrides.cassandra.storage.storageClass }}' }, tags: ['validate-input', 'validate-input-k8s'] }
```
Comment (anaik91) [inline playbook.yaml:29]: @RiflerRick : No need to explicitly send the variable as it is accessible in the python scripts 

As you can do something like below in the task

```
- name: execute validate kubernetes platform script
  block:
    - script: |
        validate_k8s_objects.py \
        --input_data '{{ overrides_string }}' \
        --generate_certificates '{{ generate_certificates }}' \
        --create_service_account '{{ create_service_account }}' \
        --cassandra_storage_class '{{ overrides.cassandra.storage.storageClass }}'
   ```

File: playbook.yaml [hunk 2]
```diff
@@ -26,14 +26,15 @@
...
     - { role: cert-manager, tags: ['cert-manager'] }
     - { role: certificates, vars: { generate_certificates: true }, tags: ['certs'] }
     - { role: prepare-helm, tags: ['prepare-helm'] }
     - { role: prepare-service-accounts, tags: ['prepare-service-accounts'] }
     - { role: enable-apigee-synchronizer, tags: ['enable-synchronizer'] }
     - { role: bootstrap-apigee-crds, tags: ['bootstrap-apigee-crds'] }
     - { role: apigee-hybrid-overrides, tags: ['generate-overrides'] }
+    - { role: apigee-custom-sc, vars: { storageClassName: '{{ overrides.cassandra.storage.storageClass }}', storageClassType: '{{ overrides.cassandra.storage.storageClassProperties.provisionerType }}', scParameters: '{{ overrides.cassandra.storage.storageClassProperties.parameters }}' }, tags: ['create-custom-sc'] }
```
Comment (anaik91) [inline playbook.yaml:37]: @RiflerRick : similar to https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31/files#r1441612629

File: playbook.yaml [hunk 3]
```diff
@@ -111,7 +112,7 @@
   roles:
     - { role: validate-input, tags: ['validate-input'] }
     - { role: validate-input-apigee-control-plane, tags: ['validate-input', 'validate-input-apigee'] }
-    - { role: validate-input-k8s-cluster, tags: ['validate-input', 'validate-input-k8s'] }
+    - { role: validate-input-k8s-cluster, vars: { custom_cassandra_storage_class: '{{ overrides.cassandra.storage.storageClass }}' }, tags: ['validate-input', 'validate-input-k8s'] }
```
Comment (anaik91) [inline playbook.yaml:115]: @RiflerRick : same as https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31/files#r1441612629

File: playbook.yaml [hunk 4]
```diff
@@ -124,6 +125,7 @@
     - { role: bootstrap-apigee-crds, tags: ['bootstrap-apigee-crds'] }
     - { role: get-cassandra-seed, vars: { kubeconfig: "{{ kubeconfigs.primary }}" }, tags: ['cass-seed'] }
     - { role: apigee-hybrid-overrides, vars: { cassandra_seed_host: "{{ cassandra_seed_data }}"}, tags: ['generate-overrides'] }
+    - { role: apigee-custom-sc, vars: { storageClassName: '{{ overrides.cassandra.storage.storageClass }}', storageClassType: '{{ overrides.cassandra.storage.storageClassProperties.provisionerType }}', scParameters: '{{ overrides.cassandra.storage.storageClassProperties.parameters }}' }, tags: ['create-custom-sc'] }
```
Comment (anaik91) [inline playbook.yaml:128]: @RiflerRick : same as https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31/files#r1441612629

File: roles/apigee-custom-sc/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+# tasks file for apigee-custom-sc role
+
+- name: Create custom storage class if not present
+  kubernetes.core.k8s:
+    state: present
+    template: templates/sc.yaml.j2
```
Comment (anaik91) [inline roles/apigee-custom-sc/tasks/main.yml:20]: @RiflerRick : does this automatically render the jinja template ?

File: roles/apigee-hybrid-overrides/templates/overrides.yaml.j2 [hunk 6]
```diff
@@ -128,7 +128,7 @@ cassandra:
   heapNewSize: {{ overrides.cassandra.heapNewSize }}
   storage:
     capacity: {{ overrides.cassandra.storage.capacity }}
-    storageclass: {{ overrides.cassandra.storage.storageclass }}
+    storageClass: {{ overrides.cassandra.storage.storageClass }}
```
Comment (anaik91) [inline roles/apigee-hybrid-overrides/templates/overrides.yaml.j2:131]: @RiflerRick : this should be `storageclass ` 

Refer: https://cloud.google.com/apigee/docs/hybrid/v1.10/config-prop-ref

**Review by anaik91 — CHANGES_REQUESTED** [review]
Please check comments

File: roles/apigee-hybrid-overrides/templates/overrides.yaml.j2 (hunk 6 above)
Comment (anaik91) [reply roles/apigee-hybrid-overrides/templates/overrides.yaml.j2:131]: https://cloud.google.com/apigee/docs/hybrid/v1.11/cassandra-production#configure-cassandra-storage-settings

Comment (RiflerRick) [reply roles/apigee-hybrid-overrides/templates/overrides.yaml.j2:131]: I have modified the vars file, its camelcase everywhere

File: playbook.yaml (hunk 1 above)
Comment (RiflerRick) [reply playbook.yaml:29]: sure

File: playbook.yaml (hunk 2 above)
Comment (RiflerRick) [reply playbook.yaml:37]: sure

Comment (RiflerRick) [reply playbook.yaml:37]: actually, moving the global variable references inside the role will tightly couple it. if we decide to change the structure of the variable in the vars file, we will have to start changing the roles too. The role IMO should have a clean interface. Let me know what you think
@anaik91

File: playbook.yaml (hunk 3 above)
Comment (RiflerRick) [reply playbook.yaml:115]: @anaik91 check out: https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31#discussion_r1443650869

File: playbook.yaml (hunk 4 above)
Comment (RiflerRick) [reply playbook.yaml:128]: @anaik91  check out https://github.com/apigee/ansible-apigee-hybrid-accelerator/pull/31#discussion_r1443650869

**Review by RiflerRick — COMMENTED** [review]
@anaik91 left some comments, can you please check

**Review by anaik91 — APPROVED** [review]
LGTM

**Pull request merged**